charts/ccsm-helm/test/integration/integration_test.go

```diff
@@ -0,0 +1,136 @@
+// +build integration
+
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"context"
+	"testing"
+
+	"github.com/camunda-cloud/zeebe/clients/go/pkg/zbc"
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type integrationTest struct {
+	suite.Suite
+	chartPath   string
+	release     string
+	namespace   string
+	kubeOptions *k8s.KubectlOptions
+}
+
+func TestIntegration(t *testing.T) {
+	t.Parallel()
+
+	chartPath, err := filepath.Abs("../../")
+	require.NoError(t, err)
+
+	namespace := "ccsm-helm-" + strings.ToLower(random.UniqueId())
+
+	// currently enforce the test context to be the Zeebe team's cluster, but could be change to some CI cluster
+	kubeOptions := k8s.NewKubectlOptions("gke_zeebe-io_europe-west1-b_zeebe-cluster", "", namespace)
+
+	suite.Run(t, &integrationTest{
+		chartPath:   chartPath,
+		release:     "zeebe-cluster-helm-it",
+		namespace:   namespace,
+		kubeOptions: kubeOptions,
+	})
+}
+
+func (s *integrationTest) SetupTest() {
+	k8s.CreateNamespace(s.T(), s.kubeOptions, s.namespace)
+}
+
+func (s *integrationTest) TearDownTest() {
+	k8s.DeleteNamespace(s.T(), s.kubeOptions, s.namespace)
+}
+
+func (s *integrationTest) TestGatewayConnection() {
+	// given
+	options := &helm.Options{
+		KubectlOptions: s.kubeOptions,
+	}
+
+	// when
+	helm.Install(s.T(), options, s.chartPath, s.release)
+	// cannot add the delete part in the tear down since the options are per test
+	defer helm.Delete(s.T(), options, s.release, true) // tbh I don't think it is necessary since we delete the namespace
+
+	// then
+	pods := k8s.ListPods(s.T(), s.kubeOptions, v1.ListOptions{LabelSelector: "app=camunda-cloud-self-managed"})
+
+	for _, pod := range pods {
+		k8s.WaitUntilPodAvailable(s.T(), s.kubeOptions, pod.Name, 10, time.Duration(10 * time.Second))
+	}
```

Comment:
:thought_balloon: I'm wondering if this might be unnecessarily brittle since pods might come and go (especially on this cluster with preemptible nodes). OTOH waiting for pods would be helpful to diagnose failed tests since waiting for a gateway connection might fail for a number of reasons.

Reply:
Yeah for me it was the best condition to first verify whether the installation works at all. Lets see how often it really happens that preemption kills our tests, we can always improve on them. It is I think good enough for the first iteration